src/behaviors/focusTrap.ts

```diff
@@ -71,7 +71,7 @@ export function focusTrap(
   // is not found, blur the recently-focused element so that focus doesn't leave the
   // trap zone.
   function ensureTrapZoneHasFocus(focusedElement: EventTarget | null) {
-    if (focusedElement instanceof HTMLElement) {
+    if (focusedElement instanceof HTMLElement && document.contains(container)) {
```

Comment:
Is the condition below more robust (i.e. supports either Light DOM or Shadow DOM)?

```suggestion
    if (focusedElement instanceof HTMLElement && focusedElement.getRootNode().contains(container)) {
```

[Documentation for Node.getRootNode()](https://developer.mozilla.org/en-US/docs/Web/API/Node/getRootNode)